File: JCE/dsa/9c.c
#include <stdio.h>

void insertionSort(int list1[], int n) {
    int i, j, value;
    for (i = 1; i < n; i++) {
        value = list1[i];
        j = i - 1;
        while (j >= 0 && list1[j] > value) {
            list1[j + 1] = list1[j];
            j = j - 1;
        }
        list1[j + 1] = value;
    }
}

int main() {
    int n, i;
    printf("Enter the number of elements in the array: ");
    scanf("%d", &n);

    int list1[n];
    for (i = 0; i < n; i++) {
        printf("Enter the number to be inserted in the array: ");
        scanf("%d", &list1[i]);
    }

    insertionSort(list1, n);

    printf("Sorted Array in Ascending Order:\n");
    for (i = 0; i < n; i++) {
        printf("%d ", list1[i]);
    }
    printf("\n");

    return 0;
}

File: JCE/dsa/9c.py
# Creating a function for insertion
def insertion_sort(list1):
    # Outer loop to traverse through 1 to len(list1)
    for i in range(1, len(list1)):
        value = list1[i]
        # Move elements of list1[0...i-1] that are
        # greater than value to one
        # position ahead of their
        # current position
        j = i - 1
        while j >= 0 and value < list1[j]:
            list1[j + 1] = list1[j]
            j -= 1
        list1[j + 1] = value
        print("The intermediate list is: ", list1)
    return list1

# Driver code to test above
list1 = []
n = int(input("Enter the number of elements in the array: "))
for i in range(n):
    num = int(input("Enter number to be inserted in array: "))
    list1.append(num)

print("The unsorted list is : ", list1)
print("The sorted list is : ", insertion_sort(list1))
